operator_ui/src/pages/Jobs/Show.tsx

```diff
@@ -1,98 +1,27 @@
+import React from 'react'
 import { v2 } from 'api'
 import { Route, RouteComponentProps, Switch } from 'react-router-dom'
-import Content from 'components/Content'
-import JobRunsList from 'components/JobRuns/List'
-import TaskList from 'components/Jobs/TaskList'
-import React from 'react'
-import { GWEI_PER_TOKEN } from 'utils/constants'
-import formatMinPayment from 'utils/formatWeiAsset'
-import { formatInitiators } from 'utils/jobSpecInitiators'
+import { useErrorHandler } from 'hooks/useErrorHandler'
+import { useLoadingPlaceholder } from 'hooks/useLoadingPlaceholder'
+import { JobData } from './sharedTypes'
 import { JobsDefinition } from './Definition'
 import { JobsErrors } from './Errors'
+import { RecentRuns } from './RecentRuns'
 import { RegionalNav } from './RegionalNav'
-import { ApiResponse, PaginatedApiResponse } from '@chainlink/json-api-client'
-import { JobSpec, JobRun } from 'core/store/models'
-import { useErrorHandler } from 'hooks/useErrorHandler'
-import { useLoadingPlaceholder } from 'hooks/useLoadingPlaceholder'
-
-export type JobData = {
-  jobSpec?: ApiResponse<JobSpec>['data']
-  recentRuns?: PaginatedApiResponse<JobRun[]>['data']
-  recentRunsCount: number
-}
-
-const totalLinkEarned = (job: NonNullable<JobData['jobSpec']>) => {
-  const zero = '0.000000'
-  const unformatted =
-    job.attributes.earnings &&
-    (job.attributes.earnings / GWEI_PER_TOKEN).toString()
-  const formatted =
-    unformatted &&
-    (unformatted.length >= 3 ? unformatted : (unformatted + '.').padEnd(8, '0'))
-  return formatted || zero
-}
-
-const chartCardStyles = (theme: Theme) =>
-  createStyles({
-    wrapper: {
-      marginLeft: theme.spacing.unit * 3,
-      marginTop: theme.spacing.unit * 2,
-      marginBottom: theme.spacing.unit * 2,
-    },
-    paymentText: {
-      color: theme.palette.secondary.main,
-      fontWeight: 450,
-    },
-    earnedText: {
-      color: theme.palette.text.secondary,
-      fontSize: theme.spacing.unit * 2,
-    },
-  })
-
-interface ChartProps extends WithStyles<typeof chartCardStyles> {
-  jobSpec: NonNullable<JobData['jobSpec']>
-}
-
-const ChartArea = withStyles(chartCardStyles)(
-  ({ classes, jobSpec }: ChartProps) => (
-    <Card>
-      <Grid item className={classes.wrapper}>
-        <Typography className={classes.paymentText} variant="h5">
-          Link Payment
-        </Typography>
-        <Typography className={classes.earnedText}>
-          {totalLinkEarned(jobSpec)}
-        </Typography>
-      </Grid>
-    </Card>
-  ),
-)
 
-type Props = {
-  showJobRunsCount: number
-} & RouteComponentProps<{
+type Props = RouteComponentProps<{
   jobSpecId: string
 }>
 
 const DEFAULT_PAGE = 1
 const RECENT_RUNS_COUNT = 5
 
-export const JobsShow: React.FC<Props> = ({ match, showJobRunsCount = 5 }) => {
+export const JobsShow: React.FC<Props> = ({ match }) => {
   const [state, setState] = React.useState<JobData>({
     recentRuns: [],
     recentRunsCount: 0,
   })
```

Comment:
Is `jobSpec` missing here?

Reply:
It's undefined by default, which works for us as we want to fetch it on load.